src/app/vault/ciphers.component.ts

```diff
@@ -80,6 +86,96 @@ export class CiphersComponent extends BaseCiphersComponent implements OnDestroy
         this.didScroll = this.pagedCiphers.length > this.pageSize;
     }
 
+    async loadOrganizationNames(): Promise<Map<string, string>> {
+        const organizations = await this.userService.getAllOrganizations();
+        const organizationNames = new Map<string, string>();
+
+        if (organizations.length > 0) {
+            organizations.forEach(organization => {
+                organizationNames.set(organization.id, organization.name);
+            });
+        }
+
+        return organizationNames;
+    }
+
+    getOrganizationById(c: CipherView): string {
+        if (c.organizationId != null) {
+            return this.organizationNames.get(c.organizationId);
+        }
+        return this.i18nService.t('myVault');
+    }
+
+    sortCiphers(sortBy: string): void {
+        if (this.sortByOptions.includes(sortBy)) {
+            if (sortBy === this.sortBy) {
+                this.sortedDescending = !this.sortedDescending;
+            } else {
+                this.sortBy = sortBy;
+                this.sortedDescending = true;
+            }
+
+            const self = this;
+
+            if (sortBy === 'lastEdited') {
+                this.ciphers.sort((a: CipherView, b: CipherView) => {
+                    if (self.sortedDescending) {
+                       return new Date(b.revisionDate).valueOf() - new Date(a.revisionDate).valueOf();
+                    } else {
+                        return new Date(a.revisionDate).valueOf() - new Date(b.revisionDate).valueOf();
+                    }
+                });
+            } else if (sortBy === 'dateCreated') {
+                // TO-DO
+            } else if (sortBy === 'owner') {
+                const sortingArray = this.organizationNames;
```

Comment:
This seems unnecessary

Reply:
I'm only using the set of organization names as keys for the sorting. The ciphers is the array being sorted.

Reply:
Right you are. I think your new code is much cleaner and understand here 👍